src/fileActions.jsx

```diff
@@ -625,3 +625,31 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        // TODO: is this a good value?
+        max_read_size: 150 * 1024 * 1024,
+        external: {
+            "content-disposition": `attachment; filename=\\"${selected.name}\\"`,
+            "content-type": "application/octet-stream"
```

Comment:
let's try and add `content-length:` here.

Reply:
That's not allowed to be send over, created this Cockpit issue https://github.com/cockpit-project/cockpit/issues/19861